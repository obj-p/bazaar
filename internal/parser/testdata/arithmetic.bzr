package arithmetic

func Arithmetic() {
    var answer = 42*1 + 1 - 1
}
