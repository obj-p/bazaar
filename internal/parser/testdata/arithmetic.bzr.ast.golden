&parser.Bazaar{
  Package: &parser.PackageDecl{
    Domain: []string{
      "arithmetic",
    },
  },
  Decls: []*parser.TopLevelDecl{
    {
      Function: &parser.FunctionDecl{
        Name: "Arithmetic",
        Block: []*parser.Stmt{
          {
            Var: &parser.VarDeclStmt{
              Dest: &parser.VariableExpr{
                Name: &"answer",
              },
              Source: &parser.Expr{
                Binary: &parser.BinaryExpr{
                  Left: &parser.Expr{
                    Binary: &parser.BinaryExpr{
                      Left: &parser.Expr{
                        Binary: &parser.BinaryExpr{
                          Left: &parser.Expr{
                            Unary: &parser.UnaryExpr{
                              Primary: &parser.PrimaryExpr{
                                Literal: &parser.Literal{
                                  Number: &"42",
                                },
                              },
                            },
                          },
                          Op: token.Op(token.OpMul),
                          Right: &parser.Expr{
                            Unary: &parser.UnaryExpr{
                              Primary: &parser.PrimaryExpr{
                                Literal: &parser.Literal{
                                  Number: &"1",
                                },
                              },
                            },
                          },
                        },
                      },
                      Op: token.Op(token.OpAdd),
                      Right: &parser.Expr{
                        Unary: &parser.UnaryExpr{
                          Primary: &parser.PrimaryExpr{
                            Literal: &parser.Literal{
                              Number: &"1",
                            },
                          },
                        },
                      },
                    },
                  },
                  Op: token.Op(token.OpSub),
                  Right: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Literal: &parser.Literal{
                          Number: &"1",
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  },
}
